tests/e2e/performance/test_pvc_multi_clone_performance.py

```diff
@@ -0,0 +1,408 @@
+            )
+            log.info(
+                f"Clone number {r['Clone Num']} creation speed is {r['speed']} MB/sec."
+            )
+
+    def build_params(self):
+        log.info("Start building params")
+
+        self.params["nspace"] = self.pvc_obj.namespace
+        self.params["pvcname"] = self.pvc_obj.name
+
+        log_file_name = os.path.basename(__file__).replace(".py", ".log")
+
+        full_log = f"{ocsci_log_path()}/{log_file_name}"
+        logging.basicConfig(
+            filename=full_log, level=logging.INFO, format=constants.LOG_FORMAT
+        )
+
+        self.params["datasize"] = int(self.params["filesize"].replace("M", ""))
+
+        self.params["clone_yaml"] = constants.CSI_CEPHFS_PVC_CLONE_YAML
+        if self.interface == constants.CEPHBLOCKPOOL:
+            self.params["clone_yaml"] = constants.CSI_RBD_PVC_CLONE_YAML
+
+        output = self.run_oc_command(
+            cmd=f"get pod {self.pod_obj.name} -o yaml", namespace=self.params["nspace"]
+        )
+
+        results = yaml.safe_load("\n".join(output))
+        self.params["path"] = results["spec"]["containers"][0]["volumeMounts"][0][
+            "mountPath"
+        ]
+        log.info(f"path - {self.params['path']}")
+
+        # reading template of clone yaml file
+        with open(self.params["clone_yaml"], "r") as stream:
+            try:
+                clone_yaml = yaml.safe_load(stream)
+                clone_yaml["spec"]["storageClassName"] = self.pvc_obj.backed_sc
+                clone_yaml["spec"]["dataSource"]["name"] = self.params["pvcname"]
+                clone_yaml["spec"]["resources"]["requests"]["storage"] = (
+                    str(self.pvc_obj.size) + "Gi"
+                )
+            except yaml.YAMLError as exc:
+                log.error(f"Can not read template yaml file {exc}")
+        log.info(
+            f'Clone yaml file : {self.params["clone_yaml"]} '
+            f"Content of clone yaml file {clone_yaml}"
+        )
+
+        return clone_yaml
+
+    def get_log_names(self):
+        """
+        Finds names for log files pods in which logs for clone creation are located
+        For CephFS: 2 pods that start with "csi-cephfsplugin-provisioner" prefix
+        For RBD: 2 pods that start with "csi-rbdplugin-provisioner" prefix
+
+        """
+        self.params["logs"] = []
+
+        pods = self.run_oc_command(cmd="get pod", namespace="openshift-storage")
+        if self.params["ERRMSG"] in pods:
+            raise Exception("Can not get csi controller pod")
+
+        provisioning_name = "csi-cephfsplugin-provisioner"
+        if self.interface == constants.CEPHBLOCKPOOL:
+            provisioning_name = "csi-rbdplugin-provisioner"
+
+        for line in pods:
+            if provisioning_name in line:
+                self.params["logs"].append(line.split()[0])
+        log.info(f'The logs pods are : {self.params["logs"]}')
+
+    def run_fio_on_pod(self, file_size):
```

Comment:
@ypersky1980 @Avilir I think we should start adding all these generic functions to some common library file.
@keesturam WDYT?

Reply:
Let's create new file for scale and perf and continue adding these functions there for lesser object creations.

Reply:
@ramkiperiy  - I'll open a github issue for creating performance library and transferring generic functions and fixing the tests correspondingly. I prefer to do it in a separate PR, since anyway not only this test will need a fix, and I also prefer that we merge multi clone performance test so that it would be running, and the mext priority would be creating the library and rearranging our clone and snapshot tests. Do you agree ?

Reply:
Yes, Please create github issue.